tools/mapping/gen_domain_files/src/gen_domain.F90

```diff
@@ -470,14 +514,19 @@ end subroutine gen_domain
 
 !===========================================================================
 
-  subroutine check_ret(ret)
+  subroutine check_ret(ret, fatal)
     ! Check return status from netcdf call
     implicit none
     integer, intent(in) :: ret
+    logical, intent(in), optional :: fatal
+    logical :: fatal_local = .true.
```

Comment:
As above, I think you want to declare `fatal_local` and set it to `.true.` in two calls. In this case, my understanding is that setting the variable in the declaration like this implicitly adds the `save` attribute, so in the block

```
call check_ret(ret1, fatal=.false.)
call check_ret(ret2)
```

The second call is treated as `fatal = .false.` which is not the desired behavior.

Reply:
Good catch. This should be fixed.

Reply:
Fixed.